### PR TITLE
Add option to compute MI

### PR DESCRIPTION
**Pull request by SaraDupont**
Fix issue #1054 

- Add option to compute normalized mutual information in `sct_maths`
- move function `calc_mi` from `sct_label_vertebrae` to `sct_maths` to avoid code duplication
- output (flag `-o`) can be either a text file (`.txt`) or a pickle file (`.pkl`, `.pklz` or `.pickle`)
- Add option to compute the Pearson's R correlation coefficient between two flatten images in `sct_maths`

### CODE REVIEW

File: scripts/sct_maths.py [hunk 1]
```diff
@@ -131,6 +131,21 @@ def get_parser():
...
                       mandatory=False,
                       example="")
+
+    parser.usage.addSection("\nSimilarity metric")
+    parser.add_option(name='-mi',
+                      type_value='file',
+                      description='Compute the mutual information (MI) between both input files (-i and -mi).',
+                      mandatory=False,
+                      example="")
+    '''
```
Comment (peristeri) [inline scripts/sct_maths.py]: Is the option -corr needed? It is commented out.

Comment (SaraDupont) [inline scripts/sct_maths.py]: It's gonna be needed in the next commit ;)

File: scripts/sct_maths.py [hunk 2]
```diff
@@ -472,7 +506,65 @@ def laplacian(data, sigmas):
...
+        file_out.close()
+
+    else:
+        import pickle, gzip
+        if ext_out == '.pklz':
+            pickle.dump(res, gzip.open(fname_out, 'wb'), protocol=2)
+        else:
+            pickle.dump(res, open(fname_out, 'w'), protocol=2)
+
+def calc_MI(x, y, nbins=32, normalized=False):
```
Comment (benjamindeleener) [inline scripts/sct_maths.py]: It may be silly, but I don't like the function name. I would change `calc_MI` to `compute_MI` or even `MI`.

Comment (benjamindeleener) [inline scripts/sct_maths.py]: Same argument for correlation. Full name are better in my point of view, as they are more understandable.

Comment (SaraDupont) [inline scripts/sct_maths.py]: Yes good point !

File: scripts/sct_maths.py [hunk 3]
```diff
@@ -472,7 +506,69 @@ def laplacian(data, sigmas):
...
+        file_out.close()
+
+    else:
+        import pickle, gzip
+        if ext_out == '.pklz':
+            pickle.dump(res, gzip.open(fname_out, 'wb'), protocol=2)
+        else:
+            pickle.dump(res, open(fname_out, 'w'), protocol=2)
+
+def mutual_information(x, y, nbins=32, normalized=False):
```
Comment (peristeri) [inline scripts/sct_maths.py]: The function documentation could also use details on what is returned and what is `nbins`.
`
"""Compute the mutual information between clusters x and y.

:param x: numpy.array cluster
:param y: numpy.array cluster
:param nbins: number of bins for not normalized mutual information
:param normalized: flag if x and y are normalized 
:returns: float non-negative value 
`

If you feel that the function parameters are straight forward, it is ok to delete the param from the documentation.

File: scripts/sct_maths.py [hunk 4]
```diff
@@ -472,7 +506,69 @@ def laplacian(data, sigmas):
     # from scipy.ndimage.filters import laplace
     # return laplace(data.astype(float))
 
+def compute_similarity(data1, data2, fname_out='', metric='', verbose=1):
```
Comment (peristeri) [inline scripts/sct_maths.py]: In the case of this function, where the complexity is greater than the other 2, I would prefer some documentation.

File: scripts/sct_maths.py (hunk 3 above)
Comment (SaraDupont) [inline scripts/sct_maths.py]: Very good point !  I just addressed it in commit e00a0af.

**Pull request merged**